cron-ci/build-and-test.sh

```diff
@@ -0,0 +1,126 @@
+    prev_hash=$(cat ${HPC_HOMEDIR}/prev_hash.txt)
+    if [[ "$current_hash" == "$prev_hash" ]]; then
+        build=false
+    fi
+fi
+
+if [[ "$build" == true ]]; then
+
+    # Don't build these libs because NOAA RDHPCS systems can't use wget or connect to gitlab
+    # finds the name of the library and changes the next line to "build: NO"
+    sed -i '/^udunits/{n;s/.*/  build: NO/}' ${HPC_STACK_FILE}
+    sed -i '/^szip/{n;s/.*/  build: NO/}' ${HPC_STACK_FILE}
+    sed -i '/^nccmp/{n;s/.*/  build: NO/}' ${HPC_STACK_FILE}
+    sed -i '/^cdo/{n;s/.*/  build: NO/}' ${HPC_STACK_FILE}
+    sed -i '/^boost/{n;s/.*/  build: NO/}' ${HPC_STACK_FILE}
+    sed -i '/^eigen/{n;s/.*/  build: NO/}' ${HPC_STACK_FILE}
+
+    # pass no to setup_modules when asked for native compiler/mpi. Using modules.
+    yes no | ./setup_modules.sh -c ${HPC_CONFIG} -p ${HPC_INSTALL_PATH} >> ${hpc_log} 2>&1
+
+    ./build_stack.sh -c ${HPC_CONFIG} -p ${HPC_INSTALL_PATH} -y ${HPC_STACK_FILE} -m >> ${hpc_log} 2>&1
+
+    # Edit stack file to build debug version of ESMF
+    # Turn everything to build: NO, then enable just ESMF, then set debug: YES
+    sed -i 's/build: YES/build: NO/' ${HPC_STACK_FILE}
+    sed -i '/^esmf/{n;s/.*/  build: YES/}' ${HPC_STACK_FILE}
+    sed -i 's/debug: NO/debug: YES/' ${HPC_STACK_FILE}
+
+    # Build debug version of ESMF
+    ./build_stack.sh -c ${HPC_CONFIG} -p ${HPC_INSTALL_PATH} -y ${HPC_STACK_FILE} -m >> ${hpc_log} 2>&1
+
+    # save the current hash as the previous hash to compare to next time
+    echo $current_hash > $HPC_HOMEDIR/prev_hash.txt
+
+    if ! grep -qi "build_stack.sh: SUCCESS!" ${hpc_log}; then
+        export TEST_UFS=false
+        echo "hpc-stack did not build correctly. Disabling ufs-weather-model tests"
+    fi
+    
+   if [[ "$TEST_UFS" == true ]]; then
+
+       # mm-dd-yyy-hh:mm
+       ufs_logdate=$(date +'%m-%d-%Y-%R')
+       ufs_logname=ufs_${ufs_logdate}.log
+       ufs_log=${LOG_PATH}/${ufs_logname}
+
+       cd ${HPC_DOWNLOAD_PATH}
+       rm -rf ufs-weather-model
+
+       git clone https://github.com/ufs-community/ufs-weather-model.git >> ${ufs_log} 2>&1
+
+       cd ufs-weather-model
+       git submodule update --init --recursive >> ${ufs_log} 2>&1
+       ufs_hash=$(git rev-parse HEAD)
+       
+       # change module use to new hpc-stack location
+       sed -i "s:module use /.*/stack:module use ${HPC_INSTALL_PATH}/modulefiles/stack:" modulefiles/${HPC_MACHINE_ID}.intel/*
+
+       # remove version from module load
+       sed -i "s:module load \(.*\)/.*:module load \1:" modulefiles/${machine}.intel/*
+
+       # give version to ESMF because two versions are built
+       sed -i "s:module load esmf:module load esmf/${STACK_esmf_version}:" modulefiles/${machine}.intel/fv3
+       sed -i "s:module load esmf:module load esmf/${STACK_esmf_version}-debug:" modulefiles/${machine}.intel/fv3_debug
+
+       cd tests
+       ./rt.sh -f -e >> ${ufs_log} 2>&1
+   fi
+fi
```

Comment:
I suggest we put this (TEST_UFS) in another script specifically for UFS.
In the future, we can add as many applications to build and test e.g. JEDI.

Keep this script to build the stack alone.